src/webhook/validation/activegate.go

```diff
@@ -66,6 +81,65 @@ func missingActiveGateMemoryLimit(dv *dynakubeValidator, dynakube *dynatracev1be
 	return ""
 }
 
+func invalidActiveGateProxyUrl(dv *dynakubeValidator, dynakube *dynatracev1beta1.DynaKube) string {
+	if dynakube.Spec.Proxy != nil {
+		if len(dynakube.Spec.Proxy.ValueFrom) > 0 {
+			var proxySecret corev1.Secret
+			err := dv.clt.Get(context.TODO(), client.ObjectKey{Name: dynakube.Spec.Proxy.ValueFrom, Namespace: dynakube.Namespace}, &proxySecret)
+			if k8serrors.IsNotFound(err) {
+				return errorMissingActiveGateProxySecret
+			} else if err != nil {
+				return fmt.Sprintf("error occurred while reading PROXY secret indicated in the Dynakube specification (%s)", err.Error())
+			}
+			proxyUrl, ok := proxySecret.Data[agproxysecret.ProxySecretKey]
+			if !ok {
+				return errorInvalidProxySecretFormat
+			}
+			return validateProxyUrl(string(proxyUrl), errorInvalidProxySecretUrl, errorInvalidProxySecretEvalCharacter)
+		} else if len(dynakube.Spec.Proxy.Value) > 0 {
+			return validateProxyUrl(dynakube.Spec.Proxy.Value, errorInvalidActiveGateProxyUrl, errorInvalidEvalCharacter)
+		}
+	}
+	return ""
+}
+
 func memoryLimitSet(resources corev1.ResourceRequirements) bool {
 	return resources.Limits != nil && resources.Limits.Memory() != nil
 }
+
+// proxyUrl is valid if
+// 1) encoded
+// 2) "`\ are escaped using \
```

Comment:
does that mean it must be either escaped or encoded or must it be both?
If it must be both, would that then work with the dtclient, since it would take the password unencoded?

Reply:
my fault, characters can not be escaped